core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -562,6 +566,38 @@ public synchronized void startInternal() throws InterruptedException, IOExceptio
       throw new IOException(errorMessage, e.getCause());
     }
     LOG.info("Started Raft Journal System in {}ms", System.currentTimeMillis() - startTime);
+    joinQuorum();
+  }
+
+  private void joinQuorum() {
+    InetSocketAddress localAddress = mConf.getLocalAddress();
+    // Send a request to join the quorum.
+    // If the server is already part of the quorum, this operation is a noop.
+    AddQuorumServerRequest request = AddQuorumServerRequest.newBuilder()
+        .setServerAddress(NetAddress.newBuilder()
+            .setHost(localAddress.getHostString())
+            .setRpcPort(localAddress.getPort()))
+        .build();
+    RaftClient client = createClient();
+    client.sendReadOnlyAsync(Message.valueOf(
+        UnsafeByteOperations.unsafeWrap(
+            JournalQueryRequest
+                .newBuilder()
+                .setAddQuorumServerRequest(request)
+                .build().toByteArray()
+        ))).whenComplete((reply, t) -> {
+          if (t != null) {
+            LOG.error("Exception occurred while joining quorum", t);
+          }
+          if (reply != null && reply.getException() != null) {
+            LOG.error("Received an error while joining quorum", reply.getException());
```

Comment:
If joinQuorum failed, the master with raft journal system will still run?? Should we shut it down or retry joining?

Reply:
Yes. The join process is a best effort call with a retry. When a new quorum is forming a new leader might not be available when master starts. We should not shutdown the master for that. If the master later is rejected by the quorum because the unsuccessful join, the server will be shutdown.